presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -72,5 +73,20 @@ protected void setup(Binder binder)
         binder.install(new CachingHiveMetastoreModule());
 
         install(new ThriftMetastoreAuthenticationModule());
+
+        install(installModuleIf(
+                ThriftMetastoreConfig.class,
+                ThriftMetastoreConfig::isConnectionPoolEnabled,
+                this::usePooledHiveMetastoreClientFactory));
+    }
+
+    private void usePooledHiveMetastoreClientFactory(Binder binder)
+    {
+        newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                .setBinding().to(PooledHiveMetastoreClientFactory.class).in(Scopes.SINGLETON);
+        binder.bind(ThriftMetastoreClientFactory.class)
+                .annotatedWith(PooledHiveMetastoreClientFactory.ForPooledMetastore.class)
+                .to(DefaultThriftMetastoreClientFactory.class)
```

Comment:
Using `DefaultThriftClientFactory` here makes it not possible to easily replace implementation via `OptionalBinder` defined above.

Maybe instead do sth like this:

```java
    @Retention(RUNTIME)
    @Target({FIELD, PARAMETER, METHOD})
    @Qualifier
    public @interface Delegate {}

    @Override
    protected void setup(Binder binder)
    {
        newOptionalBinder(binder, Key.get(ThriftMetastoreClientFactory.class, Delegate.class))
                .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);

        install(installModuleIf(
                ThriftMetastoreConfig.class,
                thriftMetastoreConfig -> !thriftMetastoreConfig.isConnectionPoolEnabled(),
                this::useNonPooledHiveMetastoreClientFactory));

        install(installModuleIf(
                ThriftMetastoreConfig.class,
                thriftMetastoreConfig -> thriftMetastoreConfig.isConnectionPoolEnabled(),
                this::usePooledHiveMetastoreClientFactory));
   ....
   }


    private void useNonPooledHiveMetastoreClientFactory(Binder binder)
    {
        binder.bind(ThriftMetastoreClientFactory.class)
                .to(Key.get(ThriftMetastoreClientFactory.class, Delegate.class))
                .in(Scopes.SINGLETON);
    }

    private void usePooledHiveMetastoreClientFactory(Binder binder)
    {
        newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
                .setBinding().to(PooledHiveMetastoreClientFactory.class).in(Scopes.SINGLETON);
        binder.bind(ThriftMetastoreClientFactory.class)
                .annotatedWith(PooledHiveMetastoreClientFactory.ForPooledMetastore.class)
                .to(Key.get(ThriftMetastoreClientFactory.class, Delegate.class))
                .in(Scopes.SINGLETON);
    }
```

Now we should be able to have pooling working with overriden base thrift connection factory by overriding binding on optional binder for `Key.get(ThriftMetastoreClientFactory.class, Delegate.class)`